**103.Поясните «взаимоотношение» исключений и деструкторов.**  
При генерации исключения из-за раскрутки стека вызываются деструкторы локальных объектов.
Если в этот момент сгенерируется новое исключение это приведет к вызову terminate()
